File: src/main/java/no/nav/fo/veilarbregistrering/registrering/BrukerRegistreringType.java
package no.nav.fo.veilarbregistrering.registrering;


import no.nav.fo.veilarbregistrering.metrics.Metric;

public enum BrukerRegistreringType implements Metric {
    ORDINAER("ORDINAER"),
    SYKMELDT("SYKMELDT");

    private String brukerRegistreringType;

    BrukerRegistreringType(String brukerRegistreringType) {
        this.brukerRegistreringType = brukerRegistreringType;
    }

    @Override
    public String toString() {
        return brukerRegistreringType;
    }

    @Override
    public String fieldName() {
        return "type";
    }

    @Override
    public String value() {
        return this.toString();
    }
}

File: src/main/kotlin/no/nav/fo/veilarbregistrering/registrering/BrukerRegistreringType.kt
package no.nav.fo.veilarbregistrering.registrering

import no.nav.fo.veilarbregistrering.metrics.Metric

enum class BrukerRegistreringType : Metric {
    ORDINAER, SYKMELDT;

    override fun fieldName(): String {
        return "type"
    }

    override fun value(): String {
        return this.toString()
    }
}